Sources/Datadog/CrashReporting/Integrations/CrashReportingIntegration.swift

```diff
@@ -4,7 +4,70 @@
  * Copyright 2019-Present Datadog, Inc.
  */
 
-/// An integration for sending crash reports to Datadog.
+/// An integration for sending crash reports.
 internal protocol CrashReportingIntegration {
+    /// Send the crash report et context to integrations.
+    ///
+    /// - Parameters:
+    ///   - report: The crash report.
+    ///   - context: The crash context
     func send(report: DDCrashReport, with context: CrashContext)
 }
+
+/// An integration for sending crash reports to Datadog Core.
+internal struct CrashReportingCoreIntegration: CrashReportingIntegration {
+    /// Defines keys referencing Crash Report message on the bus.
+    internal enum MessageKeys {
+        /// The key for a crash message.
+        static let crash = "crash"
+
+        /// The key for a crash log message.
+        static let crashLog = "crash-log"
+    }
+
+    /// The core for sending crash report and context.
+    let core: DatadogCoreProtocol
+
+    /// Send the crash report et context on the bus of the core.
+    ///
+    /// - Parameters:
+    ///   - report: The crash report.
+    ///   - context: The crash context
+    func send(report: DDCrashReport, with context: CrashContext) {
+        guard context.trackingConsent == .granted else {
+            return // Only authorized crash reports can be send
+        }
+
+        sendRUM(
+            baggage: [
+                "report": report,
+                "context": context
+            ]
+        )
+    }
+
+    private func sendRUM(baggage: FeatureBaggage) {
+        core.send(
+            message: .custom(key: MessageKeys.crash, baggage: baggage),
+            else: { self.sendLog(baggage: baggage) }
+        )
+    }
+
+    private func sendLog(baggage: FeatureBaggage) {
+        core.send(
+            message: .custom(key: MessageKeys.crashLog, baggage: baggage),
+            else: printError
+        )
+    }
+
+    private func printError() {
+        // This case is not reachable in higher abstraction but we add sanity warning.
+        DD.logger.error(
+            """
+            In order to use Crash Reporting, RUM or Logging feature must be enabled.
+            Make sure `.enableRUM(true)` or `.enableLogging(true)` are configured
+            when initializing Datadog SDK.
+            """
+        )
+    }
```

Comment:
This log won't print at initialisation like before, but after processing a crash. Customers will have less chance of seeing this. But with decoupled modules, I don't see other options..

Reply:
Before, the user was seeing this just after initialisation, so they could see any misconfiguration right away. Now, they will see this after processing a crash, which mean after the app crash and was restarted with Xcode.

Reply:
Ah, got it 👍. I think it still makes sense to print this error here (e.g. for someone who is troubleshooting their "crashes not appearing in DD"). Somewhen later, we may think of some other mean of "early spotting misconfigurations" in V2 (depends on where do we land on with public API design).